include/deal.II/dofs/dof_handler.h

```diff
@@ -26,6 +26,8 @@
 #include <deal.II/base/iterator_range.h>
 #include <deal.II/base/smartpointer.h>
 
+#include <deal.II/distributed/cell_data_transfer.templates.h>
+#include <deal.II/distributed/shared_tria.h>
```

Comment:
Do you need these includes here with the forward declaration below?

I'm sure you will only need them in `dofs/dof_handler.cc`.